drivers/hyperv/hyperv.go

```diff
@@ -447,7 +486,44 @@ func (d *Driver) GetIP() (string, error) {
 		return "", fmt.Errorf("IP not found")
 	}
 
-	return resp[0], nil
+	switch d.PreferredNetworkProtocol {
+	case PreferIPv4:
+		for _, ipStr := range resp {
+			ip := net.ParseIP(ipStr)
+			if isIPv4(ipStr) && ip.To4() != nil && ip.IsGlobalUnicast() {
+				return ipStr, nil
+			}
+		}
+
+	case PreferIPv6:
+		for _, ipStr := range resp {
+			ip := net.ParseIP(ipStr)
+			if !isIPv4(ipStr) && ip.IsGlobalUnicast() {
+				return ipStr, nil
+			}
+		}
+
+	default:
+		var preferredIP string
+		for _, ipStr := range resp {
+			ip := net.ParseIP(ipStr)
+			if ip.IsGlobalUnicast() {
+				if preferredIP == "" {
```

Comment:
this would mean that the LAST ip address, that is shared on the switch, will be returned.
So essentially, LIFO (Last in First out)...

When 10.0.0.1, 172.16.0.1 and 192.168.0.1 is given to an interface, we return 192.168.0.1 and not 10.0.0.1 which might serve a larger range?

Reply:
The reason I bring this up, is that this seems changed behaviour from previously. That would not be the intention of a 'default' fallback.

Reply:
I don't think it acts much differently than the original one.

There are two differences before this change

1) Global Unicast will always be excluded

This is intended to exclude fe80:: for IPv6 for most of the cases when I found that fe80:: will be always assigned to the interfaces when IPv6 is enabled. When it tries to wait for SSH connectivity it might end up use this address.
For IPv4 only 169.254 range will be excluded.

2) Prefer IPv4 over IPv6 in the default case (which may bring some slight confusion on the PreferIPv4 / PreferIPv6 /Default and it seems like to use IPv4Only / IPv6Only / Default would be better)
This is also been seen in other drivers as well.

When there are IPv4 addresses assigned for the interfaces, it will always the first IPv4 address available based on the addresses reported by Hyper-V. So if the interface says 10.0.0.1, 172.16.0.1, 192.168.0.1, it will return 10.0.0.1. Only when there is no IPv4 addreses found it will return the first IPv6 Global Unique address.